CMSgov/mint-app
# [EASI-3713] - Part 1 - FrequencyForm Component and anticipatedPaymentFrequency Implementation

# EASI-3713

## Changes and Description

This PR is part 1 of all the frequency updates to forms.  The subsequent PRs will work of the groundwork laid here.
The first implementation is complete for  anticipatedPaymentFrequency question - [Figma](https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?type=design&node-id=7327-41926&mode=design&t=dlbdWco7FrBdF5O2-0)

- Created `FrequencyForm` component to be used in any frequency type form
- Added implementation of `FrequencyForm` to `Payments.anticipatedPaymentFrequency`
- Updated types to use placeholder `FrequencyTypeNew` as the standard/generic type for all frequency questions
- Added two util functions `formatListItems` and `formatListOtherItems` to maintain correct data order when rendering on the readonly view
- Updated tests



## How to test this change

- Navigate to page 2 of 3 for payments (where anticipatedPaymentFrequency question is rendered)
- Verify correct render and form functionality according to Figma
- Verify the data persists and renders correctly in the readonly view
- Verify the data renders in the CSV

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
